SU2_CFD/src/solver_adjoint_discrete.cpp

```diff
@@ -329,10 +329,41 @@ void CDiscAdjSolver::RegisterVariables(CGeometry *geometry, CConfig *config, boo
     config->SetTotalTemperatureIn_BC(Temperature);
   }
 
+  /*--- Register incompressible initialization values as input ---*/
 
-    /*--- Here it is possible to register other variables as input that influence the flow solution
-     * and thereby also the objective function. The adjoint values (i.e. the derivatives) can be
-     * extracted in the ExtractAdjointVariables routine. ---*/
+  if ((config->GetKind_Regime() == INCOMPRESSIBLE) &&
```

Comment:
Finally ;) Thanks for adding that. I always wanted to change that ...

Reply:
Of course! It is very useful to be able to add these things. It is an initial implementation geared toward internal flows, but could potentially be improved.